docs/sources/clients/promtail/configuration.md

```diff
@@ -930,6 +933,55 @@ labels:
 [use_incoming_timestamp: <bool> | default = false]
 ```
 
+### GCP Log
+
+The `gcplog` block configures Promtail receive GCP logs. It has two possible strategies for doing so, depending on how the 
+`subscription_type` attribute is configured:
+1. **Pull**: Using GCP PubSub [pull subscriptions](https://cloud.google.com/pubsub/docs/pull). Promtail will consume log messages directly from the configured GCP PubSub topic.
+2. **Push**: Using GCP PubSub [push subscriptions](https://cloud.google.com/pubsub/docs/push). Promtail will expose an HTTP server, to which GCP will deliver logs.
+
+When picking the `push` subscription type, keep in mind the following:
+- The `server` configuration is the same as [server](#server), since Promtail exposes an HTTP server for target that requires so.
+- An endpoint at `POST /gcp/api/v1/push`, which expects requests from GCP PubSub message delivery system.
+
+```yaml
+# Type of subscription used to fetch logs from GCP. Can be either `pull` (default) or `push`.
+[subscription_type: <string> | default = "pull"]
+
+# [pull] GCP project ID
+[project_id: <string>]
+
+# [pull] GCP PubSub subscription from where Promtail will pull logs from
+[subscription: <string>]
+
+# [push] Server configuration options
+[server: <server_config>]
```

Comment:
Since we're using the square brackets ([]) to indicate optional configuration parameters, I'd like to eliminate them from the description prose.  My preference is to state the subscription_type in the sentence, as in this suggestion:
```suggestion
# If the subscription_type is pull,  the GCP project ID
[project_id: <string>]

# If the subscription_type is pull, GCP PubSub subscription from where Promtail will pull logs from
[subscription: <string>]

# If the subscription_type is push, the server configuration options
[server: <server_config>]
```